plugins/caddy/main.go

```diff
@@ -20,36 +20,50 @@ import (
 )
 
 type key string
+
 const getterContextCtxKey key = "getter_context"
+const moduleName = "souin_cache"
+const moduleID caddy.ModuleID = "http.handlers." + moduleName
 
 func init() {
 	caddy.RegisterModule(SouinCaddyPlugin{})
-	httpcaddyfile.RegisterGlobalOption("souin", parseCaddyfileGlobalOption)
-	httpcaddyfile.RegisterHandlerDirective("souin", parseCaddyfileHandlerDirective)
+	httpcaddyfile.RegisterGlobalOption(moduleName, parseCaddyfileGlobalOption)
+	httpcaddyfile.RegisterHandlerDirective(moduleName, parseCaddyfileHandlerDirective)
 }
 
-var staticConfig Configuration
+var (
+	staticConfig Configuration
```

Comment:
Having a global `staticConfig` isn't right. There's no way to configure global defaults from JSON config with your current setup.

The right way to do it is with a Caddy app. A Caddy app is a essentially a top-level module which acts like a singleton. So to implement this, you'd make a new struct like `SouinApp` which holds the global defaults (and don't forget the `json` tags). Then in your HTTP handler module (i.e. `SouinCaddyPlugin`) in `Provision`, you can call `ctx.App("souin_cache")` to load the Caddy app from config and grab the config values from there.

Then in `parseCaddyfileGlobalOption`, you should return a `httpcaddyfile.App`, which tells the Caddyfile adapter to take the result as the config for the Caddy app. See https://github.com/abiosoft/caddy-exec/pull/8/files for an example of how that looks.

Reply:
That's already done in the actual cache-handler.
But okay I'll update this tonight and provide a fix to use the App struct.

Reply:
> That's already done in the actual cache-handler.

No, it's not possible to configure defaults globally from JSON, so that you can avoid configuring those per-handler, because the only place where you set it up is in `parseCaddyfileGlobalOption`, which only gets called in the Caddyfile adapter, and not JSON.